src/declaration.c

```diff
@@ -1555,6 +1555,10 @@ void VarDeclaration::checkNestedReference(Scope *sc, Loc loc)
 
 int VarDeclaration::isDataseg()
 {
+    if (ident == Id::ctfe)
+    {
+        return true;
+    }
```

Comment:
Except this, why is this needed?

Reply:
because otherwise the error handling below this is triggered.

Reply:
Doesn't that indicate that something is missing though?  This variable should never be written to the data segment.

Reply:
it won't be.
Because it does not exist.
e2ir transforms the variable into a literal `0`.

Reply:
Maybe better make it assert in that case?

Reply:
Where would I assert ? `__ctfe` is not a real variable hence it does not have complete type information.
Because it does not have complete type information it would error in `isDataseg` if we would not intercept it.